engine/src/main/scala/cromwell/engine/workflow/lifecycle/materialization/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -227,21 +228,51 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
           case Some(other) => s"Unknown version '$other' for workflow language '$languageName'".invalidNel
           case _ => language.allVersions.head._2.valid
         }
-        errorOrParse(factory).flatMap(_.validateNamespace(sourceFiles, workflowOptions, engineIoFunctions, importLocalFilesystem, serviceRegistryActor, workflowIdForLogging))
+        errorOrParse(factory).flatMap(_.validateNamespace(sourceFiles, workflowOptions, importLocalFilesystem, workflowIdForLogging))
       case Some(other) => fromEither[IO](s"Unknown workflow type: $other".invalidNelCheck[ValidatedWomNamespace])
       case None => fromEither[IO]("Need a workflow type here !".invalidNelCheck[ValidatedWomNamespace])
     }
+
     val labelsValidation: Parse[Labels] = fromEither[IO](validateLabels(sourceFiles.labelsJson).toEither)
 
     for {
       labels <- labelsValidation
       _ <- publishLabelsToMetadata(id, labels)
       validatedNamespace <- namespaceValidation
+      _ <- pushImportedFilesToMetadata(validatedNamespace.importedFileContent)
+      _ <- pushWfInputsToMetadata(validatedNamespace.womValueInputs)
       _ <- pushWfNameMetadataService(validatedNamespace.executable.entryPoint.name)
       ewd <- fromEither[IO](buildWorkflowDescriptor(id, sourceFiles, validatedNamespace, workflowOptions, labels, conf, pathBuilders).toEither)
     } yield ewd
   }
 
+  private def pushWfInputsToMetadata(workflowInputs: Map[OutputPort, WomValue]): Parse[Unit] = {
```

Comment:
ToL This function and the subsequence one could have no side effects if it just returned `PutMetadataAction` and left the actor sending to the caller.

Also no need to wrap it in the `Parse` effect.